src/main/kotlin/blackjack/BlackjackApplication.kt

```diff
@@ -1,68 +1,75 @@
 package blackjack
 
 import blackjack.model.CardDeck
+import blackjack.model.GameResult
 import blackjack.model.Player
+import blackjack.model.Players
 import blackjack.view.InputView
 import blackjack.view.OutputView
+import blackjack.view.impl.ConsoleInputView
+import blackjack.view.impl.StandardOutputView
 
-class BlackjackApplication {
+class BlackjackApplication(
+    private val cardDeck: CardDeck = CardDeck.defaultDeck(),
+    private val inputView: InputView = ConsoleInputView(),
+    private val outputView: OutputView = StandardOutputView()
+) {
     fun play() {
-        val playerNames = InputView.readPlayers()
-        val cardDeck = CardDeck.defaultDeck()
+        val playerNames = inputView.readPlayers()
+        val players = initPlayers(playerNames)
+        val dealer = initPlayer(DEALER_NAME)
+        outputView.printInitCards(dealer, players)
 
-        val players = initPlayers(playerNames, cardDeck)
-        OutputView.printInitCards(players)
+        playBlackjackGame(players)
+        drawDealerCardOrNot(dealer)
+        outputView.printCardResult(dealer, players)
 
-        playBlackjackGame(
-            players, cardDeck,
-            InputView.readPlayerPickAnswer,
-            OutputView.printPlayerCards
-        )
-
-        OutputView.printResult(players)
+        val gameResult = GameResult.of(dealer, players)
+        outputView.printGameResult(players, gameResult)
     }
 
-    fun initPlayers(names: String, deck: CardDeck): List<Player> {
-        return splitPlayerNames(names).map { name ->
-            Player(name, deck.drawCards(INIT_CARD_COUNT))
+    private fun drawDealerCardOrNot(dealer: Player) {
+        if (dealer.getFinalScore() > DEALER_DRAW_SCORE) {
+            return
         }
+
+        dealer.addCard(cardDeck.drawCard())
+        outputView.printDealerDraw(dealer)
+    }
+
+    private fun initPlayers(names: String): Players {
+        return Players(splitNames(names).map { initPlayer(it) })
+    }
+
+    private fun initPlayer(name: String): Player {
+        return Player(name, cardDeck.drawCards(INIT_CARD_COUNT))
     }
```

Comment:
`Players`, `Player`에서 팩터리 메서드로 제공되어도 괜찮을것 같아요.

```suggestion
    private fun initPlayers(names: String): Players {
        return Players.init(names, cardDeck, INIT_CARD_COUNT)
    }
```